students/ArchWu/Session3/list_lab.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+fruits = ["Apples", "Pears", "Oranges", "Peaches"]
+
+def series1(fruits):
+    answer = input("Please pick a fruit to add > ")
+    fruits.append(answer)
+    print(fruits)
+    answer = input("Please pick an integer > ")
+    while not answer.isdigit() or int(answer) > len(fruits):
+        answer = input("Please pick an integer > ")
+    print(fruits[int(answer) - 1])
+    answer = input("Please pick a fruit to add > ")
+    new_fruit = [answer]
+    fruits = new_fruit + fruits
+    print(fruits)
+    answer = input("Please pick a fruit to add > ")
+    fruits.insert(0, answer)
+    print(fruits)
+    for f in fruits:
+        if f[0] == 'P': print(f)
+
+def series2(fruits):
+    print(fruits)
+    fruits.pop()
+    print(fruits)
+    answer = input("Please pick a fruit to delete > ")
+    if answer in fruits:
+        fruits.remove(answer)
+    #bonus
+    new_fruits = fruits * 2
+    answer = input("Please pick a fruit to delete(bonus) > ")
+    while answer not in new_fruits:
+         answer = input("Please pick a fruit to delete(bonus) > ")
+    for f in fruits:
+        if f == answer:
+            new_fruits.remove(f)
+
+def series3(fruits):
+
+    for f in fruits[:]:
```

Comment:
You will want to be careful here when iterating through fruits. If you remove a fruit, the iterator runs into an issue and will skip a fruit. Try to address this issue. Hint: you may want to choose a different way to keep track of indices.

Reply:
Yea actually during Chris's office hour he pointed out this issue and let me solve it with making a copy of fruit list by`for f in fruits[:]:` than the `for f in fruits:`